src/Controller/HomeController.php

```diff
@@ -21,4 +18,33 @@ public function index()
     {
         return $this->twig->render('Home/index.html.twig');
     }
+
+    public function send()
+    {
+        $errors = [];
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $data = array_map('trim', $_POST);
+            if (empty($data['lastname'])) {
+                $errors['lastname'] = "Veuillez remplir le champ nom";
+            }
+            if (empty($data['firstname'])) {
+                $errors['firstname'] = "Veuillez remplir le champ prénom";
+            }
+            if (empty($data['mail'])) {
+                $errors['mail'] = "Veuillez remplir le champ email";
+            } elseif (!filter_var($data['mail'], FILTER_VALIDATE_EMAIL)) {
+                $errors['mail'] = "Format invalide d'email";
+            }
+            if (empty($data['message'])) {
+                $errors['message'] = "Veuillez remplir le champ message";
+            }
+            if (!empty($errors)) {
+                return $this->twig->render('Home/index.html.twig', [
```

Comment:
non
si tout s'est bien passé et que error est empty, tu ne fais pas un render, sinon tu ne "vide" pas ton POST (et pb si tu réactualises). Il faut que tu faire une redirection via header Location (et dans une future US, tu pourras ajouter l'envoie du mail juste avant)

tu enlèves le return, la méthode revoie donc :void, tu peux aussi la passer en private

et depuis index, tu appelle la méthode $this->send(), en gardant le render de index et en lui passant success éventuellement.

sinon quand tu vas rajouter d'autres choses dans index(), tu seras obligé de dupliquer dans send() ce qui n'est pas bien. On en discute si c'est pas clair